### PR TITLE
Refactor: merge notices that have same entities

### PR DESCRIPTION
**Pull request by robertu7**
**Changes**
* merge notice types that share the same entities; (breaking changes)
* deprecate legacy notice types;
* break down `enums.ts`;

**TODO**
* add circle related notices;

### CODE REVIEW

**Review by guoliu — APPROVED** [review]
🍄💪

File: src/common/utils/notice.ts [hunk 1]
```diff
@@ -1,106 +1,41 @@
...
-  article_new_subscriber: true,
-  article_new_comment: true,
-  article_mentioned_you: true,
-  subscribed_article_new_comment: true,
-  upstream_article_archived: true,
-  downstream_article_archived: true,
-  comment_pinned: true,
-  comment_new_reply: true,
-  comment_mentioned_you: true,
-  official_announcement: false,
```
Comment (devformatters) [inline src/common/utils/notice.ts:48]: This is `false`.

**Review by devformatters — APPROVED** [review]
Go. 🧥

File: src/common/utils/notice.ts (hunk 1 above)
Comment (robertu7) [reply src/common/utils/notice.ts:48]: nice catch!

**Pull request merged**